### PR TITLE
feat: onTapOutside handler

### PR DESCRIPTION
**Pull request by agufagit**
added onTapOutside, `focusNode.addListener` doesn't work when tap on another button

### CODE REVIEW

File: pubspec.lock [hunk 1]
```diff
@@ -0,0 +1,164 @@
+# Generated by pub
```
Comment (lzhuor) [inline pubspec.lock:1]: Is this pubspec.lock update necessary?

File: lib/src/auto_size_text_field.dart [hunk 2]
```diff
@@ -239,8 +239,7 @@ class AutoSizeTextField extends StatefulWidget {
   ///  * [AdaptiveTextSelectionToolbar], which is built by default.
   final EditableTextContextMenuBuilder? contextMenuBuilder;
 
-  static Widget _defaultContextMenuBuilder(
-      BuildContext context, EditableTextState editableTextState) {
+  static Widget _defaultContextMenuBuilder(BuildContext context, EditableTextState editableTextState) {
```
Comment (lzhuor) [inline lib/src/auto_size_text_field.dart:242]: Wondering if this linting and auto fix on the lengths of lines is a personal style only in your IDE or the default style from [Dart Lint ](https://pub.dev/packages/lint)?

**Review by lzhuor — COMMENTED** [review]
Thank you for enhancing it @agufagit ! Appreciate it. Left 2 minor comments.

File: pubspec.lock (hunk 1 above)
Comment (agufagit) [reply pubspec.lock:1]: it's auto generated, you can ignore it

File: lib/src/auto_size_text_field.dart (hunk 2 above)
Comment (agufagit) [reply lib/src/auto_size_text_field.dart:242]: my personal style, you can ignore it

File: pubspec.yaml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 name: auto_size_text_field
 description: Flutter TextField widget that automatically resizes text field to fit perfectly within its bounds.
-version: 2.2.2
+version: 2.3.2
```
Comment (lzhuor) [inline pubspec.yaml:3]: @agufagit could you please revert this version bump? I will handle it when I release the new version. Thank you!

**Comment by lzhuor** [comment]
Thank you @agufagit! The code LGTM 🚀 

Could you please rebase the code from the master? There is also a Nit comment to revert the version change which I will handle when I release your changes. ❤️

File: pubspec.yaml (hunk 3 above)
Comment (agufagit) [reply pubspec.yaml]: done

**Comment by andyskw** [comment]
Oh, seeing this in the next release would be so wonderful! <3! :)

**Comment by lzhuor** [comment]
@andyskw I will take a look

**Comment by ooluseye16** [comment]
Would love if this is available on the pub.dev version

**Comment by lzhuor** [comment]
@ooluseye16 @andyskw @agufagit This is published on [pub.dev in v2.2.4](https://pub.dev/packages/auto_size_text_field). 

Thank you! ❤️ Appreciate it if you could like the package on pub.dev and star it on GitHub to show some support.

**Pull request merged**